test/lru_hash_test.rb

```diff
@@ -2,6 +2,7 @@
 
 class TestLRUHash < Minitest::Test
   def setup
+    Semian.thread_safe = true
```

Comment:
This state is going to leak. You could cache the value in `@old_thread_safe` and set it in a `def teardown`. This is a nitpick.